crates/fuel-core/src/schema/tx/assemble_tx.rs

```diff
@@ -954,6 +955,15 @@ where
             }
 
             let remaining_input_slots = self.remaining_input_slots()?;
+            if remaining_input_slots == 0 {
```

Comment:
Why do we need to handle it here separately if `.coins(` returns an error later?